src/test/librbd/test_librbd.cc

```diff
@@ -6890,6 +6934,10 @@ TEST_F(TestLibRBD, DiscardAfterWrite)
   librbd::Image image;
   ASSERT_EQ(0, rbd.open(ioctx, image, name.c_str(), NULL));
 
+  if (this->is_skip_partial_discard_enabled(image)) {
```

Comment:
Nit: why does this test need to be skipped?

Reply:
This test was disabled in old codes. As it discards data at [0,256], this data is not discarded when granularity is 64k or other.

Reply:
Sorry, I had missed that it was previously excluded. For cases like this, it might make more sense to keep the original `is_skip_partial_discard_enabled` method (and the versions that take an image can re-use it) since you want to skip the test regardless of whether or not RWL is involved.

Reply:
Updated.